src/main/java/gyro/aws/cloudfront/CloudFrontOriginShield.java

```diff
@@ -0,0 +1,73 @@
+package gyro.aws.cloudfront;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+import gyro.aws.Copyable;
+import gyro.core.resource.Diffable;
+import gyro.core.resource.Updatable;
+import gyro.core.validation.Required;
+import gyro.core.validation.ValidationError;
+import software.amazon.awssdk.services.cloudfront.model.OriginShield;
+
+public class CloudFrontOriginShield extends Diffable implements Copyable<OriginShield> {
+
+    private Boolean enabled;
+    private String region;
+
+    @Required
+    @Updatable
+    public Boolean getEnabled() {
+        if (enabled == null) {
+            enabled = false;
+        }
+
+        return enabled;
+    }
+
+    public void setEnabled(Boolean enabled) {
+        this.enabled = enabled;
+    }
+
+    @Updatable
+    public String getRegion() {
+        return region;
+    }
+
+    public void setRegion(String region) {
+        this.region = region;
+    }
+
+    @Override
+    public void copyFrom(OriginShield model) {
+        setEnabled(model.enabled());
+        setRegion(model.originShieldRegion());
+    }
+
+    @Override
+    public String primaryKey() {
+        return "";
+    }
+
+    protected OriginShield toOriginShield() {
+        OriginShield.Builder builder = OriginShield.builder().enabled(getEnabled());
+
+        if (getEnabled()) {
+            builder = builder.originShieldRegion(getRegion());
+        }
+
+        return builder.build();
```

Comment:
Can the `originShieldRegion()` be set to `null`? If so, this can be reduced to a single line.

```suggestion
        return OriginShield.builder().enabled(getEnabled())
        .originShieldRegion(getEnabled() ? getRegion() : null)).build();
```